tests/unittests/config/test_schema.py

```diff
@@ -852,14 +1026,88 @@ def test_label_overrides_property_name(self):
         assert "prop1" not in meta_doc
         assert ".*" not in meta_doc
 
+    @pytest.mark.parametrize(
+        "schema,expected_doc",
+        [
+            (
+                {
+                    "properties": {
+                        "prop1": {
+                            "type": ["string", "integer"],
+                            "deprecated": True,
+                            "description": "<description>",
+                        }
+                    }
+                },
+                "**prop1:** (string/integer) DEPRECATED. <description>",
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "properties": {
+                        "prop1": {
+                            "type": ["string", "integer"],
+                            "description": "<description>",
+                            "deprecated": True,
+                        },
+                    },
+                },
+                ("**prop1:** (string/integer) DEPRECATED. <description>"),
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "$defs": {"eol_bionic": {"deprecated": True}},
+                    "properties": {
+                        "prop1": {
+                            "allOf": [
+                                {
+                                    "type": ["string", "integer"],
+                                    "description": "<description>",
+                                },
+                                {"$ref": "#/$defs/eol_bionic"},
+                            ]
+                        }
+                    },
+                },
+                ("**prop1:** (string/integer) DEPRECATED. <description>"),
+            ),
+            (
+                {
+                    "$schema": "http://json-schema.org/draft-04/schema#",
+                    "$defs": {
+                        "eol_bionic": {
+                            "deprecated": True,
+                            "description": "<description>",
+                        }
+                    },
+                    "properties": {
+                        "prop1": {
+                            "allOf": [
+                                {"type": ["string", "integer"]},
+                                {"$ref": "#/$defs/eol_bionic"},
+                            ]
+                        }
+                    },
+                },
+                ("**prop1:** (string/integer) DEPRECATED. <description>"),
+            ),
+        ],
+    )
```

Comment:
Might be worth adding an `anyOf` or `allOf` here with two different descriptions where one is deprecated and showing that the non-deprecated description shows up in docs.

Reply:
Done.